src/video_compress/cmpto_j2k.cpp

```diff
@@ -48,154 +48,658 @@
+        Codec{R10k, CMPTO_444_U10U10U10_MSB32BE_P210, VIDEO_CODEC_NONE, nullptr},
+        Codec{R12L, CMPTO_444_U12_MSB16LE_P012, RG48, R12L_to_RG48},
+};
+
+/**
+ * Exceptions for state_video_compress_j2k construction
+ */
+
+/// @brief HelpRequested Exception
+struct HelpRequested : public std::exception {
+        HelpRequested() = default;
+};
+
+/// @brief InvalidArgument Exception
+struct InvalidArgument : public std::exception {
+        InvalidArgument() = default;
+};
+
+/// @brief UnableToCreateJ2KEncoderCTX Exception
+struct UnableToCreateJ2KEncoderCTX : public std::exception {
+        UnableToCreateJ2KEncoderCTX() = default;
+};
+
+/// @brief Struct for options for J2K Compression Usage
+struct opts {
+        const char *label;
+        const char *key;
+        const char *description;
+        const char *opt_str;
+        const bool is_boolean;
+};
+
+#ifdef HAVE_CUDA
+constexpr opts cuda_opts[2] = {
+        {"Mem limit", "mem_limit", "CUDA device memory limit (in bytes), default: " TOSTRING(DEFAULT_CUDA_MEM_LIMIT), ":mem_limit=", false},
+        {"Tile limit", "tile_limit", "Number of tiles encoded at one moment by GPU (less to reduce latency, more to increase performance, 0 means infinity). default: " TOSTRING(DEFAULT_CUDA_TILE_LIMIT), ":tile_limit=", false},
+};
+constexpr opts platform_opts[1] = {
+        {"Plaform", "platform", "Platform device for the encoder to use, default: cuda", ":platform=", false},
+};
+#endif // HAVE_CUDA
+
+constexpr opts cpu_opts[2] = {
+        {"Thread count", "thread_count", "Number of threads to use on the CPU. 0 is all available. default: " TOSTRING(DEFAULT_CPU_THREAD_COUNT), ":thread_count=", false},
+        {"Image limit", "img_limit", "Number of images which can be encoded at one moment by CPU. Maximum allowed limit is thread_count. 0 is default limit. default: " TOSTRING(DEFAULT_IMG_LIMIT), ":img_limit=", false},
+};
+
+constexpr opts general_opts[5] = {
+        {"Bitrate", "quality", "Target bitrate", ":rate=", false},
+        {"Quality", "quant_coeff", "Quality in range [0-1]. default: " TOSTRING(DEFAULT_QUALITY), ":quality=", false},
+#ifdef HAVE_CUDA
+        {"Pool size", "pool_size", "Total number of frames encoder can hold at one moment. Should be greater than tile_limit or img_limit. default: " TOSTRING(DEFAULT_POOL_SIZE), ":pool_size=", false},
 #else
-using allocator = default_data_allocator;
+        {"Pool size", "pool_size", "Total number of frames encoder can hold at one moment. Should be greater than img_limit. default: " TOSTRING(DEFAULT_POOL_SIZE) , ":pool_size=", false},
+#endif
+        {"Use MCT", "mct", "Use MCT", ":mct", true},
+        {"Lossless compression", "lossless", "Enable lossless compression. default: disabled", ":lossless", true}
+};
+
+/**
+ * @fn usage
+ * @brief Display J2K Compression Usage Information
+ */
+static void usage() {
+        col() << "J2K compress platform support:\n";
+        col() << "\tCPU .... yes\n";
+#ifdef HAVE_CUDA
```

Comment:
instead of insisting on HAVE_CUDA, I think it is better to use cmpto_version::technology bit array